app/controllers/types_controller.rb

```diff
@@ -0,0 +1,63 @@
+class TypesController < ApplicationController
+  before_action :authenticate_user!
+  load_and_authorize_resource
+  before_action :set_type, only: [:show, :edit, :update, :destroy]
```

Comment:
@VanjaJeftic read my comment for the `index` method. You are most likely doubling queries by using both this and `load_and_authorize_resource`.